### ISSUE TITLE
Parameters in config.ini not implemented in v3.8.4-beta1 with v980

### ISSUE THREAD

**Issue by dvasdekis** [IbcAlpha/IBC#95]
Hi RLK (and co-authors),

I'm using the latest v3.8.4-beta1 with IBGateway v980.4j on Linux, and it appears that neither `ReadOnlyApi=no`, or `OverrideTwsApiPort=4003` get obeyed. v978 by comparison is working well.

Just a heads up!

Thank you,

Dimitri

**Comment by rlktradewright** [comment]
Can you please attach (not quote) your IBC logfile? (The whole of it.)

TWS 980.4j and IBC 3.8.4-beta.1 are working fine together on Windows (see logfile extract below). Since the jar files are identical for Windows and Linux,, it seems surprising to me that it doesn't work identically on Linux. The log file might possibly throw some light on this. If not, I'll have to test in on my Ubuntu VM, but I can't do that until after 22:00 tonight.

```
2020-09-14 11:36:18:182 IBC: Setting ReadOnlyApi
2020-09-14 11:36:18:302 IBC: Read-Only API checkbox is now set to: false
2020-09-14 11:36:18:302 IBC: Performing port configuration
2020-09-14 11:36:18:302 IBC: TWS API socket port was set to 7498
2020-09-14 11:36:18:307 IBC: TWS API socket port now set to 7497
2020-09-14 11:36:18:308 IBC: Click button: OK
```

**Comment by guoguojin** [comment]
I'm experiencing the same issues, I've attached our latest log file for your reference.

```
================================================================================

Starting IBC version 3.8.4-beta.1 on 2020-09-15 at 21:03:37

Operating system: Linux b3d92b828998 4.15.0-88-generic #88-Ubuntu SMP Tue Feb 11 20:11:34 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

Arguments:

TWS version = 980
Entry point = ibcalpha.ibc.IbcGateway
--tws-path = /home/user/Jts
--tws-settings-path =
--ibc-path = /home/user/IBController
--ibc-ini = /home/user/IBController/config.ini
--mode =
--java-path =
--user =
--pw =
--fix-user =
--fix-pw =

=================================
Generating the classpath
Classpath=/home/user/Jts/ibgateway/980/jars/jts4launch-980.jar:/home/user/Jts/ibgateway/980/jars/jxbrowser-7.5.jar:/home/user/Jts/ibgateway/980/jars/jxbrowser-linux64-7.5.jar:/home/user/Jts/ibgateway/980/jars/jxbrowser-swing-7.5.jar:/home/user/Jts/ibgateway/980/jars/locales.jar:/home/user/Jts/ibgateway/980/jars/log4j-api-2.12.0.jar:/home/user/Jts/ibgateway/980/jars/log4j-core-2.12.0.jar:/home/user/Jts/ibgateway/980/jars/total-2020.jar:/home/user/Jts/ibgateway/980/jars/twslaunch-980.jar:/home/user/Jts/ibgateway/980/jars/twslaunch-install4j-1.10.jar:/home/user/IBController/IBC.jar

Generating the JAVA VM options
Java VM Options=-Xmx768m -XX:+UseG1GC -XX:MaxGCPauseMillis=200 -XX:ParallelGCThreads=20 -XX:ConcGCThreads=5 -XX:InitiatingHeapOccupancyPercent=70

Determining the location of java executable
Location of java executable=/home/user/.i4j_jres/1.8.0_152-tzdata2019c_64/bin

Starting Gateway with this command:
"/home/user/.i4j_jres/1.8.0_152-tzdata2019c_64/bin/java" -cp "/home/user/Jts/ibgateway/980/jars/jts4launch-980.jar:/home/user/Jts/ibgateway/980/jars/jxbrowser-7.5.jar:/home/user/Jts/ibgateway/980/jars/jxbrowser-linux64-7.5.jar:/home/user/Jts/ibgateway/980/jars/jxbrowser-swing-7.5.jar:/home/user/Jts/ibgateway/980/jars/locales.jar:/home/user/Jts/ibgateway/980/jars/log4j-api-2.12.0.jar:/home/user/Jts/ibgateway/980/jars/log4j-core-2.12.0.jar:/home/user/Jts/ibgateway/980/jars/total-2020.jar:/home/user/Jts/ibgateway/980/jars/twslaunch-980.jar:/home/user/Jts/ibgateway/980/jars/twslaunch-install4j-1.10.jar:/home/user/IBController/IBC.jar" -Xmx768m -XX:+UseG1GC -XX:MaxGCPauseMillis=200 -XX:ParallelGCThreads=20 -XX:ConcGCThreads=5 -XX:InitiatingHeapOccupancyPercent=70 ibcalpha.ibc.IbcGateway "/home/user/IBController/config.ini"

IBC Settings:
    AcceptIncomingConnectionAction=accept
    AcceptNonBrokerageAccountWarning=yes
    AllowBlindTrading=yes
    BindAddress=127.0.0.1
    ClosedownAt=
    CommandPrompt=
    CommandServerPort=7462
    ControlFrom=
    DismissNSEComplianceNotice=yes
    DismissPasswordExpiryWarning=no
    ExistingSessionDetectedAction=primaryoverride
    FIX=no
    FIXLoginId=***
    FIXPassword=***
    IbAutoClosedown=no
    IbDir=
    IbLoginId=***
    IbPassword=***
    LogComponents=never
    MinimizeMainWindow=no
    OverrideTwsApiPort=
    ReadOnlyApi=no
    ReadOnlyLogin=no
    SaveTwsSettingsAt=
    ShowAllTrades=no
    StoreSettingsOnServer=no
    SuppressInfoMessages=yes
    TradingMode=live
End IBC Settings

2020-09-15 21:03:37:928 IBC: Version: 3.8.4-beta.1
System Properties
------------------------------------------------------------
java.runtime.name = Java(TM) SE Runtime Environment
sun.boot.library.path = /home/user/.i4j_jres/1.8.0_152-tzdata2019c_64/lib/amd64
java.vm.version = 25.152-b16
java.vm.vendor = Oracle Corporation
java.vendor.url = http://java.oracle.com/
path.separator = :
java.vm.name = Java HotSpot(TM) 64-Bit Server VM
file.encoding.pkg = sun.io
user.country = US
sun.java.launcher = SUN_STANDARD
sun.os.patch.level = unknown
java.vm.specification.name = Java Virtual Machine Specification
user.dir = /home/user/Jts
java.runtime.version = 1.8.0_152-b16
java.awt.graphicsenv = sun.awt.X11GraphicsEnvironment
java.endorsed.dirs = /home/user/.i4j_jres/1.8.0_152-tzdata2019c_64/lib/endorsed
os.arch = amd64
java.io.tmpdir = /tmp
line.separator =

java.vm.specification.vendor = Oracle Corporation
os.name = Linux
sun.jnu.encoding = ANSI_X3.4-1968
java.library.path = /usr/java/packages/lib/amd64:/usr/lib64:/lib64:/lib:/usr/lib
java.specification.name = Java Platform API Specification
java.class.version = 52.0
sun.management.compiler = HotSpot 64-Bit Tiered Compilers
os.version = 4.15.0-88-generic
user.home = /home/user
user.timezone = Etc/UTC
java.awt.printerjob = sun.print.PSPrinterJob
file.encoding = ANSI_X3.4-1968
java.specification.version = 1.8
java.class.path = /home/user/Jts/ibgateway/980/jars/jts4launch-980.jar:/home/user/Jts/ibgateway/980/jars/jxbrowser-7.5.jar:/home/user/Jts/ibgateway/980/jars/jxbrowser-linux64-7.5.jar:/home/user/Jts/ibgateway/980/jars/jxbrowser-swing-7.5.jar:/home/user/Jts/ibgateway/980/jars/locales.jar:/home/user/Jts/ibgateway/980/jars/log4j-api-2.12.0.jar:/home/user/Jts/ibgateway/980/jars/log4j-core-2.12.0.jar:/home/user/Jts/ibgateway/980/jars/total-2020.jar:/home/user/Jts/ibgateway/980/jars/twslaunch-980.jar:/home/user/Jts/ibgateway/980/jars/twslaunch-install4j-1.10.jar:/home/user/IBController/IBC.jar
user.name = user
java.vm.specification.version = 1.8
sun.java.command = ibcalpha.ibc.IbcGateway /home/user/IBController/config.ini
java.home = /home/user/.i4j_jres/1.8.0_152-tzdata2019c_64
sun.arch.data.model = 64
user.language = en
java.specification.vendor = Oracle Corporation
awt.toolkit = sun.awt.X11.XToolkit
java.vm.info = mixed mode
java.version = 1.8.0_152
java.ext.dirs = /home/user/.i4j_jres/1.8.0_152-tzdata2019c_64/lib/ext:/usr/java/packages/lib/ext
sun.boot.class.path = /home/user/.i4j_jres/1.8.0_152-tzdata2019c_64/lib/resources.jar:/home/user/.i4j_jres/1.8.0_152-tzdata2019c_64/lib/rt.jar:/home/user/.i4j_jres/1.8.0_152-tzdata2019c_64/lib/sunrsasign.jar:/home/user/.i4j_jres/1.8.0_152-tzdata2019c_64/lib/jsse.jar:/home/user/.i4j_jres/1.8.0_152-tzdata2019c_64/lib/jce.jar:/home/user/.i4j_jres/1.8.0_152-tzdata2019c_64/lib/charsets.jar:/home/user/.i4j_jres/1.8.0_152-tzdata2019c_64/lib/jfr.jar:/home/user/.i4j_jres/1.8.0_152-tzdata2019c_64/classes
java.vendor = Oracle Corporation
file.separator = /
java.vendor.url.bug = http://bugreport.sun.com/bugreport/
sun.io.unicode.encoding = UnicodeLittle
sun.cpu.endian = little
sun.cpu.isalist =
------------------------------------------------------------
2020-09-15 21:03:37:930 IBC: Using default settings provider: ini file is /home/user/IBController/config.ini
2020-09-15 21:03:37:930 IBC: Using default login manager: getting username and password from args but not found. Will get from settings
2020-09-15 21:03:37:930 IBC: Using default main window manager: constructor parameter isGateway=true
2020-09-15 21:03:37:930 IBC: Using default trading mode manager: constructor parameter args but trading mode not present - will be taken from settings
2020-09-15 21:03:37:931 IBC: Using default config dialog manager
2020-09-15 21:03:37:937 IBC: CommandServer is starting with port 7462
2020-09-15 21:03:37:980 IBC: CommandServer listening on address: 127.0.0.1 port: 7462
2020-09-15 21:03:37:980 IBC: CommandServer started and is ready to accept commands
2020-09-15 21:03:38:016 IBC: TWS Settings directory is: /home/user/Jts
2020-09-15 21:03:38:019 IBC: Ensuring /home/user/Jts/jts.ini contains required minimal lines
2020-09-15 21:03:38:019 IBC: Found setting: [Logon]/s3store=true
2020-09-15 21:03:38:020 IBC: Found setting: [Logon]/Locale=en
2020-09-15 21:03:38:020 IBC: Found setting: [Logon]/displayedproxymsg=1
2020-09-15 21:03:38:020 IBC: Found setting: [IBGateway]/ApiOnly=true
2020-09-15 21:03:38:020 IBC: Confirmed /home/user/Jts/jts.ini contains required minimal lines
2020-09-15 21:03:38:020 IBC: Starting session: will exit if login dialog is not displayed within 30 seconds
2020-09-15 21:03:39:234 IBC: Getting config dialog
2020-09-15 21:03:39:235 IBC: Creating config dialog future
2020-09-15 21:03:39:236 IBC: Getting main window
2020-09-15 21:03:39:236 IBC: Creating main window future
2020-09-15 21:03:40:405 IBC: Detected frame entitled: Interactive Brokers Gateway; event=Activated
2020-09-15 21:03:40:406 IBC: Trading mode from settings: tradingMode=live
2020-09-15 21:03:40:407 IBC: Setting Trading mode = live
2020-09-15 21:03:40:523 IBC: Detected frame entitled: Interactive Brokers Gateway; event=Focused
2020-09-15 21:03:40:525 IBC: Detected frame entitled: Interactive Brokers Gateway; event=Opened
2020-09-15 21:03:40:625 IBC: Click button: Log In
2020-09-15 21:03:41:346 IBC: Detected frame entitled: Interactive Brokers Gateway; event=Lost focus
2020-09-15 21:03:41:349 IBC: Detected frame entitled: Loading...; event=Activated
2020-09-15 21:03:41:351 IBC: Detected frame entitled: Loading...; event=Focused
2020-09-15 21:03:41:351 IBC: Detected frame entitled: Loading...; event=Opened
2020-09-15 21:03:41:704 IBC: Detected frame entitled: Loading...; event=Lost focus
2020-09-15 21:03:41:705 IBC: Detected frame entitled: Authenticating...; event=Activated
2020-09-15 21:03:41:707 IBC: Detected frame entitled: Authenticating...; event=Focused
2020-09-15 21:03:41:707 IBC: Detected frame entitled: Authenticating...; event=Opened
LogModuleConfigurator-Init: Log4j Ver2.x found on classpath
LogModuleConfigurator-Init: LogModuleConfigurator initialized with Log4j Verd.x
2020-09-15 21:03:44:304 IBC: Detected frame entitled: Initializing environment...; event=Lost focus
2020-09-15 21:03:44:698 IBC: Detected frame entitled: Starting application...; event=Closed
2020-09-15 21:03:44:699 IBC: Login has completed
remove Client 1000

================================================================================

Starting IBC version 3.8.4-beta.1 on 2020-09-15 at 21:04:32

Operating system: Linux 45b65aa4e95d 4.15.0-88-generic #88-Ubuntu SMP Tue Feb 11 20:11:34 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

Arguments:

TWS version = 980
Entry point = ibcalpha.ibc.IbcGateway
--tws-path = /home/user/Jts
--tws-settings-path =
--ibc-path = /home/user/IBController
--ibc-ini = /home/user/IBController/config.ini
--mode =
--java-path =
--user =
--pw =
--fix-user =
--fix-pw =

=================================
Generating the classpath
Classpath=/home/user/Jts/ibgateway/980/jars/jts4launch-980.jar:/home/user/Jts/ibgateway/980/jars/jxbrowser-7.5.jar:/home/user/Jts/ibgateway/980/jars/jxbrowser-linux64-7.5.jar:/home/user/Jts/ibgateway/980/jars/jxbrowser-swing-7.5.jar:/home/user/Jts/ibgateway/980/jars/locales.jar:/home/user/Jts/ibgateway/980/jars/log4j-api-2.12.0.jar:/home/user/Jts/ibgateway/980/jars/log4j-core-2.12.0.jar:/home/user/Jts/ibgateway/980/jars/total-2020.jar:/home/user/Jts/ibgateway/980/jars/twslaunch-980.jar:/home/user/Jts/ibgateway/980/jars/twslaunch-install4j-1.10.jar:/home/user/IBController/IBC.jar

Generating the JAVA VM options
Java VM Options=-Xmx768m -XX:+UseG1GC -XX:MaxGCPauseMillis=200 -XX:ParallelGCThreads=20 -XX:ConcGCThreads=5 -XX:InitiatingHeapOccupancyPercent=70

Determining the location of java executable
Location of java executable=/home/user/.i4j_jres/1.8.0_152-tzdata2019c_64/bin

Starting Gateway with this command:
"/home/user/.i4j_jres/1.8.0_152-tzdata2019c_64/bin/java" -cp "/home/user/Jts/ibgateway/980/jars/jts4launch-980.jar:/home/user/Jts/ibgateway/980/jars/jxbrowser-7.5.jar:/home/user/Jts/ibgateway/980/jars/jxbrowser-linux64-7.5.jar:/home/user/Jts/ibgateway/980/jars/jxbrowser-swing-7.5.jar:/home/user/Jts/ibgateway/980/jars/locales.jar:/home/user/Jts/ibgateway/980/jars/log4j-api-2.12.0.jar:/home/user/Jts/ibgateway/980/jars/log4j-core-2.12.0.jar:/home/user/Jts/ibgateway/980/jars/total-2020.jar:/home/user/Jts/ibgateway/980/jars/twslaunch-980.jar:/home/user/Jts/ibgateway/980/jars/twslaunch-install4j-1.10.jar:/home/user/IBController/IBC.jar" -Xmx768m -XX:+UseG1GC -XX:MaxGCPauseMillis=200 -XX:ParallelGCThreads=20 -XX:ConcGCThreads=5 -XX:InitiatingHeapOccupancyPercent=70 ibcalpha.ibc.IbcGateway "/home/user/IBController/config.ini"

IBC Settings:
    AcceptIncomingConnectionAction=accept
    AcceptNonBrokerageAccountWarning=yes
    AllowBlindTrading=yes
    BindAddress=127.0.0.1
    ClosedownAt=
    CommandPrompt=
    CommandServerPort=7462
    ControlFrom=
    DismissNSEComplianceNotice=yes
    DismissPasswordExpiryWarning=no
    ExistingSessionDetectedAction=primaryoverride
    FIX=no
    FIXLoginId=***
    FIXPassword=***
    IbAutoClosedown=no
    IbDir=
    IbLoginId=***
    IbPassword=***
    LogComponents=never
    MinimizeMainWindow=no
    OverrideTwsApiPort=
    ReadOnlyApi=no
    ReadOnlyLogin=no
    SaveTwsSettingsAt=
    ShowAllTrades=no
    StoreSettingsOnServer=no
    SuppressInfoMessages=yes
    TradingMode=live
End IBC Settings

2020-09-15 21:04:32:212 IBC: Version: 3.8.4-beta.1
System Properties
------------------------------------------------------------
java.runtime.name = Java(TM) SE Runtime Environment
sun.boot.library.path = /home/user/.i4j_jres/1.8.0_152-tzdata2019c_64/lib/amd64
java.vm.version = 25.152-b16
java.vm.vendor = Oracle Corporation
java.vendor.url = http://java.oracle.com/
path.separator = :
java.vm.name = Java HotSpot(TM) 64-Bit Server VM
file.encoding.pkg = sun.io
user.country = US
sun.java.launcher = SUN_STANDARD
sun.os.patch.level = unknown
java.vm.specification.name = Java Virtual Machine Specification
user.dir = /home/user/Jts
java.runtime.version = 1.8.0_152-b16
java.awt.graphicsenv = sun.awt.X11GraphicsEnvironment
java.endorsed.dirs = /home/user/.i4j_jres/1.8.0_152-tzdata2019c_64/lib/endorsed
os.arch = amd64
java.io.tmpdir = /tmp
line.separator =

java.vm.specification.vendor = Oracle Corporation
os.name = Linux
sun.jnu.encoding = ANSI_X3.4-1968
java.library.path = /usr/java/packages/lib/amd64:/usr/lib64:/lib64:/lib:/usr/lib
java.specification.name = Java Platform API Specification
java.class.version = 52.0
sun.management.compiler = HotSpot 64-Bit Tiered Compilers
os.version = 4.15.0-88-generic
user.home = /home/user
user.timezone = Etc/UTC
java.awt.printerjob = sun.print.PSPrinterJob
file.encoding = ANSI_X3.4-1968
java.specification.version = 1.8
java.class.path = /home/user/Jts/ibgateway/980/jars/jts4launch-980.jar:/home/user/Jts/ibgateway/980/jars/jxbrowser-7.5.jar:/home/user/Jts/ibgateway/980/jars/jxbrowser-linux64-7.5.jar:/home/user/Jts/ibgateway/980/jars/jxbrowser-swing-7.5.jar:/home/user/Jts/ibgateway/980/jars/locales.jar:/home/user/Jts/ibgateway/980/jars/log4j-api-2.12.0.jar:/home/user/Jts/ibgateway/980/jars/log4j-core-2.12.0.jar:/home/user/Jts/ibgateway/980/jars/total-2020.jar:/home/user/Jts/ibgateway/980/jars/twslaunch-980.jar:/home/user/Jts/ibgateway/980/jars/twslaunch-install4j-1.10.jar:/home/user/IBController/IBC.jar
user.name = user
java.vm.specification.version = 1.8
sun.java.command = ibcalpha.ibc.IbcGateway /home/user/IBController/config.ini
java.home = /home/user/.i4j_jres/1.8.0_152-tzdata2019c_64
sun.arch.data.model = 64
user.language = en
java.specification.vendor = Oracle Corporation
awt.toolkit = sun.awt.X11.XToolkit
java.vm.info = mixed mode
java.version = 1.8.0_152
java.ext.dirs = /home/user/.i4j_jres/1.8.0_152-tzdata2019c_64/lib/ext:/usr/java/packages/lib/ext
sun.boot.class.path = /home/user/.i4j_jres/1.8.0_152-tzdata2019c_64/lib/resources.jar:/home/user/.i4j_jres/1.8.0_152-tzdata2019c_64/lib/rt.jar:/home/user/.i4j_jres/1.8.0_152-tzdata2019c_64/lib/sunrsasign.jar:/home/user/.i4j_jres/1.8.0_152-tzdata2019c_64/lib/jsse.jar:/home/user/.i4j_jres/1.8.0_152-tzdata2019c_64/lib/jce.jar:/home/user/.i4j_jres/1.8.0_152-tzdata2019c_64/lib/charsets.jar:/home/user/.i4j_jres/1.8.0_152-tzdata2019c_64/lib/jfr.jar:/home/user/.i4j_jres/1.8.0_152-tzdata2019c_64/classes
java.vendor = Oracle Corporation
file.separator = /
java.vendor.url.bug = http://bugreport.sun.com/bugreport/
sun.io.unicode.encoding = UnicodeLittle
sun.cpu.endian = little
sun.cpu.isalist =
------------------------------------------------------------
2020-09-15 21:04:32:214 IBC: Using default settings provider: ini file is /home/user/IBController/config.ini
2020-09-15 21:04:32:214 IBC: Using default login manager: getting username and password from args but not found. Will get from settings
2020-09-15 21:04:32:214 IBC: Using default main window manager: constructor parameter isGateway=true
2020-09-15 21:04:32:214 IBC: Using default trading mode manager: constructor parameter args but trading mode not present - will be taken from settings
2020-09-15 21:04:32:215 IBC: Using default config dialog manager
2020-09-15 21:04:32:221 IBC: CommandServer is starting with port 7462
2020-09-15 21:04:32:263 IBC: CommandServer listening on address: 127.0.0.1 port: 7462
2020-09-15 21:04:32:263 IBC: CommandServer started and is ready to accept commands
2020-09-15 21:04:32:300 IBC: TWS Settings directory is: /home/user/Jts
2020-09-15 21:04:32:303 IBC: Ensuring /home/user/Jts/jts.ini contains required minimal lines
2020-09-15 21:04:32:303 IBC: Found setting: [Logon]/s3store=true
2020-09-15 21:04:32:303 IBC: Found setting: [Logon]/Locale=en
2020-09-15 21:04:32:303 IBC: Found setting: [Logon]/displayedproxymsg=1
2020-09-15 21:04:32:303 IBC: Found setting: [IBGateway]/ApiOnly=true
2020-09-15 21:04:32:303 IBC: Confirmed /home/user/Jts/jts.ini contains required minimal lines
2020-09-15 21:04:32:303 IBC: Starting session: will exit if login dialog is not displayed within 30 seconds
2020-09-15 21:04:33:547 IBC: Getting config dialog
2020-09-15 21:04:33:547 IBC: Creating config dialog future
2020-09-15 21:04:33:548 IBC: Getting main window
2020-09-15 21:04:33:548 IBC: Creating main window future
2020-09-15 21:04:34:721 IBC: Detected frame entitled: Interactive Brokers Gateway; event=Activated
2020-09-15 21:04:34:723 IBC: Trading mode from settings: tradingMode=live
2020-09-15 21:04:34:723 IBC: Setting Trading mode = live
2020-09-15 21:04:34:847 IBC: Detected frame entitled: Interactive Brokers Gateway; event=Focused
2020-09-15 21:04:34:848 IBC: Detected frame entitled: Interactive Brokers Gateway; event=Opened
2020-09-15 21:04:34:942 IBC: Click button: Log In
2020-09-15 21:04:35:655 IBC: Detected frame entitled: Interactive Brokers Gateway; event=Lost focus
2020-09-15 21:04:35:658 IBC: Detected frame entitled: Loading...; event=Activated
2020-09-15 21:04:35:660 IBC: Detected frame entitled: Loading...; event=Focused
2020-09-15 21:04:35:660 IBC: Detected frame entitled: Loading...; event=Opened
2020-09-15 21:04:36:013 IBC: Detected frame entitled: Loading...; event=Lost focus
2020-09-15 21:04:36:014 IBC: Detected frame entitled: Authenticating...; event=Activated
2020-09-15 21:04:36:016 IBC: Detected frame entitled: Authenticating...; event=Focused
2020-09-15 21:04:36:017 IBC: Detected frame entitled: Authenticating...; event=Opened
LogModuleConfigurator-Init: Log4j Ver2.x found on classpath
LogModuleConfigurator-Init: LogModuleConfigurator initialized with Log4j Verd.x
2020-09-15 21:04:39:064 IBC: Detected frame entitled: Initializing environment...; event=Lost focus
2020-09-15 21:04:39:266 IBC: Detected frame entitled: Starting application...; event=Closed
2020-09-15 21:04:39:266 IBC: Login has completed
remove Client 1000
2020-09-15 21:06:04:020 IBC: Detected frame entitled: IB Gateway; event=Activated
2020-09-15 21:06:04:024 IBC: Detected frame entitled: IB Gateway; event=Focused
addLogConsole Client 1000
2020-09-15 21:06:10:238 IBC: Detected frame entitled: IB Gateway; event=Lost focus
2020-09-15 21:06:10:238 IBC: Detected dialog entitled: U3090334 Trader Workstation Configuration; event=Activated
2020-09-15 21:06:10:240 IBC: Detected dialog entitled: U3090334 Trader Workstation Configuration; event=Focused
2020-09-15 21:06:10:240 IBC: Detected dialog entitled: U3090334 Trader Workstation Configuration; event=Opened
2020-09-15 21:06:37:539 IBC: Detected dialog entitled: U3090334 Trader Workstation Configuration; event=Lost focus
2020-09-15 21:06:37:540 IBC: Detected dialog entitled: U3090334 Trader Workstation Configuration; event=Closed
2020-09-15 21:08:28:378 IBC: Detected frame entitled: IB Gateway; event=Activated
2020-09-15 21:08:28:381 IBC: Detected frame entitled: IB Gateway; event=Focused
addLogConsole Client 1000
2020-09-15 21:08:33:942 IBC: Detected dialog entitled: U4273284 Trader Workstation Configuration; event=Opened
2020-09-15 21:08:33:944 IBC: Detected frame entitled: IB Gateway; event=Lost focus
2020-09-15 21:08:33:945 IBC: Detected dialog entitled: U4273284 Trader Workstation Configuration; event=Activated
2020-09-15 21:08:33:945 IBC: Detected dialog entitled: U4273284 Trader Workstation Configuration; event=Focused
2020-09-15 21:08:54:831 IBC: Detected dialog entitled: U4273284 Trader Workstation Configuration; event=Lost focus
2020-09-15 21:08:54:832 IBC: Detected dialog entitled: U4273284 Trader Workstation Configuration; event=Closed
2020-09-15 21:22:15:124 IBC: Detected frame entitled: IB Gateway; event=Activated
2020-09-15 21:22:15:126 IBC: Detected frame entitled: IB Gateway; event=Focused
2020-09-16 02:10:00:453 IBC: Detected dialog entitled: API client needs write access action confirmation; event=Opened
2020-09-16 02:10:00:469 IBC: Detected dialog entitled: API client needs write access action confirmation; event=Activated
2020-09-16 02:10:00:470 IBC: Detected dialog entitled: API client needs write access action confirmation; event=Focused
2020-09-16 02:10:00:476 IBC: Detected dialog entitled: API client needs write access action confirmation; event=Lost focus
2020-09-16 02:10:00:486 IBC: Detected dialog entitled: API client needs write access action confirmation; event=Activated
2020-09-16 02:10:00:486 IBC: Detected dialog entitled: API client needs write access action confirmation; event=Focused
2020-09-16 04:15:43:328 IBC: Detected frame entitled: IB Gateway; event=Lost focus
2020-09-16 04:15:43:328 IBC: Detected frame entitled: Login; event=Activated
2020-09-16 04:15:43:329 IBC: Detected frame entitled: Login; event=Focused
2020-09-16 04:15:43:329 IBC: Detected frame entitled: Login; event=Opened
2020-09-16 04:15:43:347 IBC: Detected frame entitled: Login; event=Opened
2020-09-16 04:20:45:455 IBC: Detected frame entitled: Starting application...; event=Closed
2020-09-16 04:20:52:792 IBC: Detected frame entitled: Starting application...; event=Lost focus
2020-09-16 04:20:52:792 IBC: Detected frame entitled: Starting application...; event=Closed
2020-09-16 06:01:02:482 IBC: Detected dialog entitled: U3090334 Trader Workstation Configuration; event=Opened
2020-09-16 06:01:02:508 IBC: Detected dialog entitled: API client needs write access action confirmation; event=Lost focus
2020-09-16 06:01:02:509 IBC: Detected dialog entitled: API client needs write access action confirmation; event=Closed
2020-09-16 06:01:02:943 IBC: Detected dialog entitled: U3090334 Trader Workstation Configuration; event=Activated
2020-09-16 06:01:02:943 IBC: Detected dialog entitled: U3090334 Trader Workstation Configuration; event=Focused
2020-09-16 06:01:09:191 IBC: Detected dialog entitled: U3090334 Trader Workstation Configuration; event=Lost focus
2020-09-16 06:01:09:192 IBC: Detected dialog entitled: U3090334 Trader Workstation Configuration; event=Closed
2020-09-16 06:02:02:425 IBC: Detected frame entitled: IB Gateway; event=Activated
2020-09-16 06:02:02:426 IBC: Detected frame entitled: IB Gateway; event=Focused
2020-09-16 06:02:03:468 IBC: Detected frame entitled: IB Gateway; event=Lost focus
2020-09-16 06:02:03:468 IBC: Detected dialog entitled: U4273284 Trader Workstation Configuration; event=Activated
2020-09-16 06:02:03:469 IBC: Detected dialog entitled: U4273284 Trader Workstation Configuration; event=Focused
2020-09-16 06:02:03:470 IBC: Detected dialog entitled: U4273284 Trader Workstation Configuration; event=Opened
2020-09-16 06:02:13:056 IBC: Detected dialog entitled: U4273284 Trader Workstation Configuration; event=Lost focus
2020-09-16 06:02:13:056 IBC: Detected dialog entitled: U4273284 Trader Workstation Configuration; event=Closed
2020-09-16 06:18:45:134 IBC: Detected frame entitled: IB Gateway; event=Activated
2020-09-16 06:18:45:135 IBC: Detected frame entitled: IB Gateway; event=Focused
2020-09-16 06:18:46:654 IBC: Detected frame entitled: IB Gateway; event=Lost focus
2020-09-16 06:18:46:655 IBC: Detected dialog entitled: U4273284 Trader Workstation Configuration; event=Activated
2020-09-16 06:18:46:657 IBC: Detected dialog entitled: U4273284 Trader Workstation Configuration; event=Focused
2020-09-16 06:18:46:658 IBC: Detected dialog entitled: U4273284 Trader Workstation Configuration; event=Opened

null(javax.swing.JRootPane){Object.Component.Container.JComponent.JRootPane}
|   null.glassPane(javax.swing.JPanel){Object.Component.Container.JComponent.JPanel}
|   null.layeredPane(javax.swing.JLayeredPane){Object.Component.Container.JComponent.JLayeredPane}
|   |   null.contentPane(javax.swing.JPanel){Object.Component.Container.JComponent.JPanel}
|   |   |   null(javax.swing.JToolTip){Object.Component.Container.JComponent.JToolTip}


2020-09-16 06:18:51:151 IBC: Detected window: type=javax.swing.Popup$HeavyWeightWindow; event=Opened
2020-09-16 06:19:25:492 IBC: Detected dialog entitled: U4273284 Trader Workstation Configuration; event=Lost focus
2020-09-16 06:19:25:493 IBC: Detected window: type=javax.swing.Popup$HeavyWeightWindow; event=Closed
2020-09-16 06:19:25:493 IBC: Detected dialog entitled: U4273284 Trader Workstation Configuration; event=Closed
2020-09-16 06:19:34:454 IBC: Detected frame entitled: IB Gateway; event=Activated
2020-09-16 06:19:34:455 IBC: Detected frame entitled: IB Gateway; event=Focused

```

**Comment by rlktradewright** [comment]
@guoguojin 

_[You didn't attach your file, you quoted it! That's exactly what I asked not to do. So now every time someone looks at this issue they have to scroll all the way past your logfile contents, which is tedious._

_To attach a file, click on the box below the one you type your reply in - where it says 'Attach files by dragging & dropping, selecting or pasting them'.]_

Yes, I'm aware there is an issue here on Linux. I spent some time last night investigating it, but I haven't yet nailed down the cause. As soon as I do, I'll fix it. But I can only do this investigation, and test a fix, at times when I don't need my paper-trading TWS running, which is mostly late at night or at the weekend.

So please be patient...

**Comment by guoguojin** [comment]
@rlktradewright apologies, fixed previous comment.

Appreciate your efforts. Not trying to rush you, just trying to help and provide a little more information.

Regards.

**Comment by rlktradewright** [comment]
This is now fixed in release 3.8.4-beta..2

Apologies for the delay in getting this done.
